### DISCUSSION TITLE
Allow App-Interfaces for more Data-types

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#3589] [Feature Requests]
Hey everyone,
Is there any reason why the App-Interface are that much restricted to specific data-types?
E.g you can't use dropdowns or radio-buttons for anything else than string values.

I think it would be pretty useful for other datatypes with limited options too (e.g for numeric values instead of the slider)

**Comment by benhaynes** [comment]
It might be an oversight, or there may be some other limitation that @rijkvanzanten can comment on. I agree that number types should be allowed — any others that seem to be missing? It should be as easy as adding them to the interface's supported type array.

**Reply by Dominic-Marcelino** [reply]
Quickly looked through the coded and I've been only missing `dropdown` and `radio-buttons` booth for following data-types:

`'string', 'text', 'boolean', 'integer', 'decimal', 'float', 'bigInteger', 'dateTime', 'date', 'time', 'timestamp'`

**Reply by benhaynes** [reply]
@rijkvanzanten any issues with adding these?

**Reply by rijkvanzanten** [reply]
Yeah, we'll have to make the interface options for those interfaces type-aware before we can open that up, otherwise the options will save the value with the wrong type (f.e. `"5.6"` instead of `5.6` for numbers). It'll also be very easy to enter an invalid value in the options as there's no validation whatsoever (f.e. entering `2021/01/05` instead of `2021-01-05`)

**Comment by Dominic-Marcelino** [comment]
@rijkvanzanten I created a first draft therefore:

https://github.com/directus/directus/compare/main...Dominic-Marcelino:feature/support-selection-interfaces

It dynamically sets the input-field-type for the value in the dropdown form.

As commented in the code the solution of cloning the returned interface is probably not the nicest, but the only one that came into my mind to change it 100% safely.

Also I added an idea of how an optional validator could be added, to not only rely on the browser. It could be build on top of [validateJs](https://validatejs.org/) for basic types. Next some custom regexp and error-messages could also be passed (compared to Vuetify). The custom error-message could be displayed directly under the input-field for example.

Both Ideas are open for discussion and can be continued.

**Comment by MarcusMBX** [comment]
It would be helpful if dropdown is also available for many-to-one relations. If the related table has a few entries, it's easier to select one from a dropdown.
I believe there was an option in directus 8, which allowed to set kind of threshold "below x show dropdown, above show dialog popup". That would be very nice in directus 9.

**Reply by MarcusMBX** [reply]
I found out, that it's possible to set "selectMode":"dropdown" in the json located in directus_fields.options
I guess there's is work in progress and setting this in database is a simple workaround meanwhile.

**Reply by rijkvanzanten** [reply]
As per https://github.com/directus/directus/discussions/4044#discussioncomment-375847, we might actually move to a setup where you choose dropdown first, and then choose whether or not to use a "raw string" or a relational value. That should clear some of this confusion up as well

**Reply by benhaynes** [reply]
Hmm, partially. If you choose a dropdown as an interface, but then your dataset grows too large for it... v8 had a threshold for auto-switching to a modal (drawer in v9). I wonder if we need something like that, or we just expect the user to switch the interface itself. 🤔

**Reply by MarcusMBX** [reply]
I believe it would be helpful to have this threshold back again because some users might be able to handle 25 entries in a dropdown, others may get confused with 10 already.
This might also depend on the use-case, if a row contains a lot of info, the modal should be shown earlier.
Long story short, it would be cool if the threshold can be adjusted. :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
